lib/ansible/modules/remote_management/manageiq/manageiq_alerts.py

```diff
@@ -0,0 +1,363 @@
+          from: "example@example.com"
+    resource_type: ContainerNode
+    expression:
+        eval_method: hostd_log_threshold
+        mode: internal
+        options: {}
+    enabled: true
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Add an alert with a "miq expression" to ManageIQ
+  manageiq_alerts:
+    state: present
+    description: Test Alert 02
+    options:
+      notifications:
+        email:
+          to: ["example@example.com"]
+          from: "example@example.com"
+    resource_type: Vm
+    expression_type: miq
+    expression:
+        and:
+          - CONTAINS:
+              tag: Vm.managed-environment
+              value: prod
+          - not:
+            CONTAINS:
+              tag: Vm.host.managed-environment
+              value: prod
+    enabled: true
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+
+- name: Delete an alert from ManageIQ
+  manageiq_alerts:
+    state: absent
+    description: Test Alert 01
+    manageiq_connection:
+      url: 'http://127.0.0.1:3000'
+      username: 'admin'
+      password: 'smartvm'
+      verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+# TODO: I copied this from yaacov's ansible/ansible PR 31233. It should be removed
+# from here once that PR is merged.
+def find_collection_resource_or_fail(module, manageiq, collection_name, **params):
+    """ Searches the collection resource by the collection name and the param passed.
+
+    Returns:
+        the resource as an object if it exists in manageiq, Fail otherwise.
+    """
+    resource = manageiq.find_collection_resource_by(collection_name, **params)
+    if resource:
+        return resource
+    else:
+        msg = "{collection_name} where {params} does not exist in manageiq".format(
+            collection_name=collection_name, params=str(params))
+    module.fail_json(msg=msg)
+
+
+class ManageIQAlert(object):
+    """ Represent a ManageIQ alert
+    """
+    def __init__(self, alert):
+        self.description = alert['description']
+        self.db = alert['db']
+        self.enabled = alert['enabled']
+        self.options = alert['options']
+        self.hash_expression = None
+        self.miq_expressipn = None
+
+        if 'hash_expression' in alert:
+            self.hash_expression = alert['hash_expression']
+        if 'miq_expression' in alert:
+            self.miq_expression = alert['miq_expression']
+            if 'exp' in self.miq_expression:
+                # miq_expression is a field that needs a special case, because
+                # it's returned surrounded by a dict named exp even though we don't
+                # send it with that dict.
```

Comment:
So this class can be initialized with both format we send and format we receive?
Neat! :heart:  Worth mentioning in docstring IMHO.

Reply:
Done